src/cookbooks/clients_last_seen_bits.md

```diff
@@ -0,0 +1,136 @@
+# Working with Bit Patterns in Clients Last Seen
+
+An powerful feature of the `clients_last_seen` methodology is that we can capture
+per-day usage information for the entire 28-day period that each row in the table
+describes. We encode usage information in "bit patterns" where the physical
+type of the field is a BigQuery 64-bit integer, but logically the integer
+represents an array of bits, with each 1 indicating a day where the given clients
+was active and each 0 indicating a day where the client was inactive.
+
+The representation is space-efficient and actually easier to manipulate
+in BigQuery than a date array would be, but requires some practice to interact
+with directly. For well-established usage and retention metrics, we generally
+provide logic in views to insulate users from having to ever consider the
+raw bit patterns, but new investigations may require novel retention definitions.
+
+## Usage: Backward-looking windows
+
+The simplest application of usage bit patterns is for calculating metrics in
+backward-looking windows. This is what we do for our canonical _usage_ measures
+DAU, WAU, and MAU.
+
+Let's imagine a single row from `clients_last_seen` with `submission_date = 2020-01-28`.
+The `days_seen_bits` field records usage for a single client over a period of 28 days
```

Comment:
Any usage criteria baked in here? Does that limit this approach?

Reply:
I will need to add some discussion of usage criteria, but yes usage criteria have to be fixed as part of the `clients_last_seen` where each usage criterion is its own bit pattern field. It's a known deficiency right now that we don't have a defined workflow for how DS can quickly test out new usage criterion definitions and propose them for inclusion. We need to do a full backfill to add a new usage criterion.

Reply:
I like the discussion above.  I would call out the challenges to add new usage criteria as noted.